core/src/main/java/io/smallrye/concurrency/SmallRyeConcurrencyManager.java

```diff
@@ -107,6 +107,9 @@ public ThreadContextProviderPlan getProviderPlan(String[] propagated, String[] u
         // check for existence
         Set<ThreadContextProvider> propagatedProviders = new HashSet<>();
         for (String type : propagatedSet) {
+            if (type.isEmpty()) {
```

Comment:
These checks are needed because spec says you can, via MP conf, pass in basically empty array which will end up as empty String here.